src/seg009.c

```diff
@@ -796,6 +796,7 @@ int __pascal far set_joy_mode() {
 	if (SDL_NumJoysticks() < 1) {
 		is_joyst_mode = 0;
 	} else {
+		SDL_GameControllerAddMappingsFromFile(gamecontrollerdb_file);
```

Comment:
This needs to check for an actual set value in gamecontrollerdb_file first, I think? i.e. something like:

```
if (*gamecontrollerdb_file)
    SDL_GameControllerAddMappingsFromFile(gamecontrollerdb_file);

```

Reply:
Ok, I've changed this now.  I went with `if (gamecontrollerdb_file[0] != '\0') ...` - is that OK?  I'm happy to change to your suggestion if you prefer.

I've tested out the following scenarios:

1. gamecontrollerdb_file option not set in config - my controller doesn't work, but the game seems to function nromally
2. gamecontrollerdb_file set to a file that exists - my controller works
3. gamecontrollerdb_file set to a file that doesn't exist - same outcome as 1.